diff -up bluez-5.50/Makefile.obexd.0003~ bluez-5.50/Makefile.obexd
--- bluez-5.50/Makefile.obexd.0003~	2018-06-10 23:01:00.266559105 +0200
+++ bluez-5.50/Makefile.obexd	2018-06-10 23:02:25.163563755 +0200
@@ -1,12 +1,12 @@
 if SYSTEMD
 systemduserunitdir = @SYSTEMD_USERUNITDIR@
 systemduserunit_DATA = obexd/src/obex.service
+endif
 
 dbussessionbusdir = @DBUS_SESSIONBUSDIR@
 dbussessionbus_DATA = obexd/src/org.bluez.obex.service
-endif
 
-EXTRA_DIST += obexd/src/obex.service.in obexd/src/org.bluez.obex.service
+EXTRA_DIST += obexd/src/obex.service.in obexd/src/org.bluez.obex.service.in
 
 if OBEX
 
diff -up bluez-5.50/obexd/src/org.bluez.obex.service.0003~ bluez-5.50/obexd/src/org.bluez.obex.service
diff -up bluez-5.50/obexd/src/org.bluez.obex.service.in.0003~ bluez-5.50/obexd/src/org.bluez.obex.service.in
--- bluez-5.50/obexd/src/org.bluez.obex.service.in.0003~	2018-06-10 23:01:00.267559105 +0200
+++ bluez-5.50/obexd/src/org.bluez.obex.service.in	2018-06-10 23:01:00.266559105 +0200
@@ -0,0 +1,4 @@
+[D-BUS Service]
+Name=org.bluez.obex
+Exec=@libexecdir@/obexd
+SystemdService=dbus-org.bluez.obex.service
