--- bluez-5.27/src/error.c.omv~	2015-01-18 13:03:51.868048646 +0100
+++ bluez-5.27/src/error.c	2015-01-18 13:05:01.330764685 +0100
@@ -40,7 +40,7 @@ DBusMessage *btd_error_invalid_args(DBus
 DBusMessage *btd_error_invalid_args_str(DBusMessage *msg, const char *str)
 {
 	return g_dbus_create_error(msg, ERROR_INTERFACE ".InvalidArguments",
-									str);
+								"%s", str);
 }
 
 DBusMessage *btd_error_busy(DBusMessage *msg)
@@ -99,7 +99,7 @@ DBusMessage *btd_error_not_authorized(DB
 
 DBusMessage *btd_error_not_permitted(DBusMessage *msg, const char *str)
 {
-	return g_dbus_create_error(msg, ERROR_INTERFACE ".NotPermitted", str);
+	return g_dbus_create_error(msg, ERROR_INTERFACE ".NotPermitted", "%s", str);
 }
 
 DBusMessage *btd_error_no_such_adapter(DBusMessage *msg)
